### PR TITLE
Add explicit support for labeled tuples

### PR DESCRIPTION
**Pull request by NathanReb**
This PR adds tests for ocp-indent's behaviour with the labeled tuples syntax (introduced in OCaml 5.4: https://github.com/ocaml/ocaml/pull/13498).

In expressions and patterns, ocp-indent was already correctly handling labeled tuples and no update were required.

In type expressions though it didn't as the type expression syntax for labels was too closely tied to arrows which resulted in the incorrect indentation of the following example:
```ocaml
a:
int
*
b:
int
```
which was misintepreted as `a: (int * b:int)`, assuming the label `a` was a function argument label.

Properly handling this isn't a trivial fix since determining how to indent it ultimately depends on whether there is an arrow up ahead.

e.g.:
```ocaml
a:
int
*
b:
int
```
should be indented as follows:
```ocaml
a:
  int
*
b:
  int
```
but
```ocaml
a:
int
*
b:
int
-> unit
```
should be indented as follows:
```ocaml
a:
  int
  *
  b:
    int
-> unit
```
to reflect `->` and `*` priorities/associativity with the label.

`ocp-indent` doesn't handle backtracking and avoids lookahead as much as possible. I came to the conclusion that in this specific case, the lookahead was unavoidable so I tried to:
1. Avoid it as much as possible by eliminating unambiguously tuple cases: a label directly past a `*` cannot be associated with a `->`
2. Limit the lookahead search range and assume no arrow is coming up past it. In theory, the code could be arbitrarily long between a label and the arrow its associated with + the code can be incomplete.

I'll happily update and improve this approach based on user feedback. My gut feeling is that in practice this will work reasonably well though it might fail us in some corner cases.

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -6,6 +6,7 @@
   code using some predefined identifiers (`TEST`, `BENCH`, `IFDEF`)
   (#345, @NathanReb)
 + Add support for starred style comments (#347, @NathanReb)
++ Add epxlicit support for labeled tuples syntax (#349, @NathanReb)
```
Comment (AltGr) [inline CHANGELOG.md]:
```suggestion
+ Add explicit support for labeled tuples syntax (#349, @NathanReb)
```

**Review by AltGr — APPROVED** [review]
Seems pretty reasonable ; the depth limit to 100 might be a bit high, but that shouldn't affect anything. Thanks!
